### PR TITLE
Add the xml color system support for ARK

### PR DESCRIPTION
**Pull request by yefeidd**
## Overview
Add the xml color system support for ARK

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## In scope
- The theme color of xml color system 
- The special color of xml color system
- The sample demo of xml color system
- The wiki of the xml color system

## Out of scope
- The compose color system

## Check list
1. [ ] Does your submission pass tests?
2. [x] Have you lint your code locally prior to submission?
3. [ ] Have you update the wiki of your change?

## Reference
- Card: https://trello.com/c/ti3FKtM7/31-3-不依赖ui-themes-1
- Related PR: 
https://github.com/TW-Smart-CoE/Android-ARK/pull/1
https://github.com/TW-Smart-CoE/Android-ARK/pull/3

### CODE REVIEW

**Comment by ssseasonnn** [comment]
nice

File: ui-sample/src/main/java/com/thoughtworks/android/ark/uisample/colorsystem/ColorSampleScreen.kt [hunk 1]
```diff
@@ -21,14 +23,38 @@ fun ColorSampleScreen() {
...
         AndroidView(factory = { ctx ->
             LayoutInflater.from(ctx).inflate(R.layout.xml_color_test, null)
         }) {
-            (it as TextView).setTextColor(ComposeColors.Dynamic.ButtonContent.colorInt(context))
+            (it as TextView).setText(R.string.special_color_test_for_xml)
+            val textColor = XmlColors.ButtonBackground.colorInt(context)
+            it.setTextColor(textColor)
         }
     }
 }
```
Comment (yunlongwen) [inline ui-sample/src/main/java/com/thoughtworks/android/ark/uisample/colorsystem/ColorSampleScreen.kt:60]: IDE configuration should be checked. `*.kt` files require a blank line.

**Review by Tony---Zhang — APPROVED** [review]

File: ui-sample/src/main/java/com/thoughtworks/android/ark/uisample/SampleUIActivity.kt [hunk 2]
```diff
@@ -4,16 +4,24 @@ import android.content.Intent
...
 import com.thoughtworks.android.ark.uisample.state.Action
 import com.thoughtworks.android.ark.uisample.state.NavigateActivityAction
-import com.thoughtworks.android.ark.ui.themes.AndroidARKTheme
 
 class SampleUIActivity : AppCompatActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContent {
-            AndroidARKTheme(isSystemInDarkTheme()) {
+            ARKTheme {
```
Comment (Tony---Zhang) [inline ui-sample/src/main/java/com/thoughtworks/android/ark/uisample/SampleUIActivity.kt:18]: Change `ARKTheme` to `AppTheme`

Comment (yefeidd) [reply ui-sample/src/main/java/com/thoughtworks/android/ark/uisample/SampleUIActivity.kt:18]: I will change it

File: ui-sample/src/main/res/layout/xml_color_system.xml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent">
+
+    <androidx.compose.ui.platform.ComposeView
+        android:id="@+id/compose_color_01"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:paddingTop="30.dp"
```
Comment (Tony---Zhang) [inline ui-sample/src/main/res/layout/xml_color_system.xml:11]: 30dp

File: ui/src/main/java/com/thoughtworks/android/ark/ui/themes/colors/APPExtendedColors.kt [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        Color(LocalContext.current.getColor(colorRes))
+
+    @Composable
+    fun alphaColor(alphaValue: Float) =
+        colorValue().let {
+            it.copy(alphaValue * it.alpha)
+        }
+
+    fun alphaColorInt(context: Context, alphaValue: Float): Int {
+        return Color(colorInt(context)).let {
```
Comment (Tony---Zhang) [inline ui/src/main/java/com/thoughtworks/android/ark/ui/themes/colors/APPExtendedColors.kt:30]: colorInt(context).let { ??

**Pull request merged**